### PR TITLE
[EventEngine] Refactoring the EventEngine Test Suite: Part 1

### PR DESCRIPTION
**Pull request by drfloob**
This is a partial refactoring, shortened to support @ctiller's work improving fix_build_deps.

Known limitations:
* redundancy in test suite configuration
* ~temporary locations for some EventEngine test utils should likely be moved into the src tree~
* A few new TODOs, to be cleaned up in Part 2+

### CODE REVIEW

File: test/core/event_engine/event_engine_test_utils.cc [hunk 1]
```diff
@@ -241,5 +217,29 @@ ConnectionManager::CreateConnection(std::string target_addr,
   return absl::CancelledError("Failed to create connection.");
 }
 
+// Returns a random message with bounded length.
+std::string GetNextSendMessage() {
```
Comment (Vignesh2208) [inline test/core/event_engine/event_engine_test_utils.cc:221]: Any specific reason why this function was moved to the end of this file from the beginning of the file ?

File: test/core/event_engine/event_engine_test_utils.h [hunk 2]
```diff
@@ -132,7 +131,38 @@ class ConnectionManager {
   std::unique_ptr<EventEngine> oracle_event_engine_;
 };
 
+void AppendStringToSliceBuffer(SliceBuffer* buf, absl::string_view data);
+
+std::string ExtractSliceBufferIntoString(SliceBuffer* buf);
```
Comment (Vignesh2208) [inline test/core/event_engine/event_engine_test_utils.h:136]: I think ExtractSliceBufferIntoString is declared twice: here and in line 136.

File: test/core/event_engine/test_suite/README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 A reusable test suite for EventEngine implementations.
 
+# Customizing tests for your EventEngine implementation
+
 To exercise a custom EventEngine, create a new bazel test target that links
 against the `//test/core/event_engine/test_suite:complete` library, and provide
```
Comment (Vignesh2208) [inline test/core/event_engine/test_suite/README.md:6]: //test/core/event_engine/test_suite:complete appears to be deleted. Does this line still hold true ?

File: test/core/event_engine/test_suite/tests/client_test.cc [hunk 4]
```diff
@@ -44,10 +44,18 @@
...
-#include "test/core/event_engine/test_suite/event_engine_test.h"
-#include "test/core/event_engine/test_suite/event_engine_test_utils.h"
+#include "test/core/event_engine/event_engine_test_utils.h"
+#include "test/core/event_engine/test_suite/event_engine_test_framework.h"
 #include "test/core/util/port.h"
 
+namespace grpc_event_engine {
+namespace experimental {
+
+void InitClientTests() {}
```
Comment (Vignesh2208) [inline test/core/event_engine/test_suite/tests/client_test.cc:54]: What is expected to go in this function implementation ?

**Review by Vignesh2208 — COMMENTED** [review]
Mostly lgtm. Left a few suggestions

File: test/core/event_engine/event_engine_test_utils.cc (hunk 1 above)
Comment (drfloob) [reply test/core/event_engine/event_engine_test_utils.cc]: No, it was a merge artifact, we had both done similar changes here. I've moved it back up near the top.

File: test/core/event_engine/event_engine_test_utils.h (hunk 2 above)
Comment (drfloob) [reply test/core/event_engine/event_engine_test_utils.h]: Good catch, fixed

File: test/core/event_engine/test_suite/README.md (hunk 3 above)
Comment (drfloob) [reply test/core/event_engine/test_suite/README.md]: Documentation updated, thanks.

File: test/core/event_engine/test_suite/tests/client_test.cc (hunk 4 above)
Comment (drfloob) [reply test/core/event_engine/test_suite/tests/client_test.cc:54]: Nothing, unfortunately. This bit of code allows `fix_build_deps` to see a source file dependency between the custom test specifications and the test suite test implementations. They were previously only associated at link-time, which is fundamentally opposed to what `fix_build_deps` is trying to do. Without this hack, the tests themselves get removed from the test suite targets.

**Review by Vignesh2208 — APPROVED** [review]

**Pull request merged**